network.go

```diff
@@ -392,7 +392,7 @@ type network interface {
 	init(config *NetworkConfig) error
 
 	// join switches the current process to the specified network namespace.
-	join(networkNSPath string) error
+	join(networkNSPath string, cb func() error) error
```

Comment:
I am fine with this patch, but then we should rename that method to `run` or `do`.